toolchain/driver/clang_runner_test.cpp

```diff
@@ -159,5 +159,40 @@ TEST(ClangRunnerTest, BuitinHeaders) {
   EXPECT_THAT(err, StrEq(""));
 }
 
+TEST(ClangRunnerTest, CompileMultipleFiles) {
+  const auto install_paths =
+      InstallPaths::MakeForBazelRunfiles(Testing::GetExePath());
+
+  // Memory leaks and other errors from running Clang can at times only manifest
+  // with repeated compilations. Use a lambda to just do a series of compiles.
+  auto compile = [&](llvm::StringRef filename, llvm::StringRef source) {
```

Comment:
Had you considered making a test class and putting this on it, to share cross-test? This looks like the same compile call as with the previous two tests, and if runner args were an argument, could possibly be shared with `DashC` too.

Reply:
I was somewhat focused on reproducing the specific failure and fixing it.

I can look at refactoring, although my initial sense is that the tests may be more clear as stand-alone things. Regardless, would prefer refactoring to be a separate PR.

Reply:
Sure, separate PR seems fine.